### PR TITLE
Need code review/worries with issue in coronasafe

### PR DESCRIPTION
**Pull request by 7tena**

### CODE REVIEW

**Comment by dauntlessnomad** [comment]
@7tena can you fix the merge conflicts

File: src/Components/Patient/ManagePatients.tsx [hunk 1]
```diff
@@ -1,20 +1,20 @@
-import loadable from '@loadable/component';
-import Box from '@material-ui/core/Box';
+import loadable from "@loadable/component";
```
Comment (riginoommen) [inline src/Components/Patient/ManagePatients.tsx]:
```suggestion
import loadable from '@loadable/component';
```

File: src/Components/Patient/ManagePatients.tsx [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
+import Box from "@material-ui/core/Box";
 import Button from "@material-ui/core/Button";
 import Grid from "@material-ui/core/Grid";
-import { makeStyles, Theme, useTheme } from '@material-ui/core/styles';
-import Typography from '@material-ui/core/Typography';
-import ArrowDownwardIcon from '@material-ui/icons/ArrowDownward';
-import { navigate, useQueryParams } from 'raviger';
-import { parsePhoneNumberFromString } from 'libphonenumber-js';
-import moment from 'moment';
+import { makeStyles, Theme, useTheme } from "@material-ui/core/styles";
```
Comment (riginoommen) [inline src/Components/Patient/ManagePatients.tsx]:
```suggestion
import { makeStyles, Theme, useTheme } from '@material-ui/core/styles';
```

File: src/Components/Patient/ManagePatients.tsx [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
+import Typography from "@material-ui/core/Typography";
```
Comment (riginoommen) [inline src/Components/Patient/ManagePatients.tsx]:
```suggestion
import Typography from '@material-ui/core/Typography';
```

File: src/Components/Patient/ManagePatients.tsx [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
+import ArrowDownwardIcon from "@material-ui/icons/ArrowDownward";
```
Comment (riginoommen) [inline src/Components/Patient/ManagePatients.tsx]:
```suggestion
import ArrowDownwardIcon from '@material-ui/icons/ArrowDownward';
```

File: src/Components/Patient/ManagePatients.tsx [hunk 5]
```diff
@@ -1,20 +1,20 @@
...
+import { navigate, useQueryParams } from "raviger";
+import { parsePhoneNumberFromString } from "libphonenumber-js";
+import moment from "moment";
```
Comment (riginoommen) [inline src/Components/Patient/ManagePatients.tsx]:
```suggestion
import { navigate, useQueryParams } from 'raviger';
import { parsePhoneNumberFromString } from 'libphonenumber-js';
import moment from 'moment';
```

File: src/Components/Patient/ManagePatients.tsx [hunk 6]
```diff
@@ -1,20 +1,20 @@
...
 import React, { useEffect, useState } from "react";
 import { CSVLink } from "react-csv";
 import { useDispatch } from "react-redux";
-import SwipeableViews from 'react-swipeable-views';
+import SwipeableViews from "react-swipeable-views";
```
Comment (riginoommen) [inline src/Components/Patient/ManagePatients.tsx]:
```suggestion
import SwipeableViews from 'react-swipeable-views';
```

File: src/Components/Patient/ManagePatients.tsx [hunk 7]
```diff
@@ -1,20 +1,20 @@
...
 import { getAllPatient } from "../../Redux/actions";
-import { PhoneNumberField } from '../Common/HelperInputFields';
-import NavTabs from '../Common/NavTabs';
+import { PhoneNumberField } from "../Common/HelperInputFields";
+import NavTabs from "../Common/NavTabs";
```
Comment (riginoommen) [inline src/Components/Patient/ManagePatients.tsx]:
```suggestion
import { PhoneNumberField } from '../Common/HelperInputFields';
import NavTabs from '../Common/NavTabs';
```

File: src/Components/Patient/ManagePatients.tsx [hunk 8]
```diff
@@ -51,24 +51,30 @@ function TabPanel(props: TabPanelProps) {
...
   root: {
     flexGrow: 1,
-    backgroundColor: theme.palette.background.paper
+    backgroundColor: theme.palette.background.paper,
   },
 }));
 
-const now = moment().format('DD-MM-YYYY:hh:mm:ss');
-
+const now = moment().format("DD-MM-YYYY:hh:mm:ss");
```
Comment (riginoommen) [inline src/Components/Patient/ManagePatients.tsx]:
```suggestion
const now = moment().format('DD-MM-YYYY:hh:mm:ss');
```

File: src/Components/Patient/ManagePatients.tsx [hunk 9]
```diff
@@ -79,33 +85,32 @@ const useStyles = makeStyles((theme) => ({
...
-
   const [data, setData] = useState([]);
   const [isLoading, setIsLoading] = useState(false);
   const [totalCount, setTotalCount] = useState(0);
   const [DownloadFile, setDownloadFile] = useState("");
   const [qParams, setQueryParams] = useQueryParams();
   const [showFilters, setShowFilters] = useState(false);
 
-  const tabValue = qParams.is_active === 'False' ? 1 : 0;
+  const tabValue = qParams.is_active === "False" ? 1 : 0;
```
Comment (riginoommen) [inline src/Components/Patient/ManagePatients.tsx]:
```suggestion
  const tabValue = qParams.is_active === 'False' ? 1 : 0;
```

File: src/Components/Patient/ManagePatients.tsx [hunk 10]
```diff
@@ -79,33 +85,32 @@ const useStyles = makeStyles((theme) => ({
...
 
   const params = {
     page: qParams.page || 1,
     name: qParams.name || undefined,
-    is_active: qParams.is_active || 'True',
+    is_active: qParams.is_active || "True",
```
Comment (riginoommen) [inline src/Components/Patient/ManagePatients.tsx]:
```suggestion
    is_active: qParams.is_active || 'True',
```

File: src/Components/Patient/ManagePatients.tsx [hunk 11]
```diff
@@ -79,33 +85,32 @@ const useStyles = makeStyles((theme) => ({
...
     disease_status: qParams.disease_status || undefined,
-    phone_number: qParams.phone_number ? parsePhoneNumberFromString(qParams.phone_number)?.format('E.164') : undefined,
+    phone_number: qParams.phone_number
+      ? parsePhoneNumberFromString(qParams.phone_number)?.format("E.164")
```
Comment (riginoommen) [inline src/Components/Patient/ManagePatients.tsx]:
```suggestion
      ? parsePhoneNumberFromString(qParams.phone_number)?.format('E.164')
```

File: src/Components/Patient/ManagePatients.tsx [hunk 12]
```diff
@@ -129,49 +137,79 @@ export const PatientManager = (props: any) => {
...
-    await handleDownload(true)
-  }
+    await handleDownload(true);
+  };
 
   useEffect(() => {
     setIsLoading(true);
 
-    dispatch(getAllPatient(params, 'listPatients'))
+    dispatch(getAllPatient(params, "listPatients"))
```
Comment (riginoommen) [inline src/Components/Patient/ManagePatients.tsx]:
```suggestion
    dispatch(getAllPatient(params, 'listPatients'))
```

File: src/Components/Patient/ManagePatients.tsx [hunk 13]
```diff
@@ -129,49 +137,79 @@ export const PatientManager = (props: any) => {
...
     setQueryParams(nParams, true);
-  }
+  };
 
   const handleTabChange = async (tab: number) => {
-
     updateQuery({
       ...qParams,
-      is_active: tab ? 'False' : 'True',
+      is_active: tab ? "False" : "True",
```
Comment (riginoommen) [inline src/Components/Patient/ManagePatients.tsx]:
```suggestion
      is_active: tab ? 'False' : 'True',
```

File: src/Components/Patient/ManagePatients.tsx [hunk 14]
```diff
@@ -217,65 +258,96 @@ export const PatientManager = (props: any) => {
         <div
           key={`usr_${patient.id}`}
           onClick={() => navigate(patientUrl)}
-          className={"w-full pb-2 cursor-pointer border-b md:flex justify-between items-center mb-3 " + (patient.disease_status == "POSITIVE" ? "bg-red-50" : "")}
+          className={
+            "w-full pb-2 cursor-pointer border-b md:flex justify-between items-center mb-3 " +
+            (patient.disease_status == "POSITIVE" ? "bg-red-50" : "")
```
Comment (riginoommen) [inline src/Components/Patient/ManagePatients.tsx]:
```suggestion
            (patient.disease_status === "POSITIVE" ? "bg-red-50" : "")
```

File: src/Components/Patient/ManagePatients.tsx [hunk 15]
```diff
@@ -217,65 +258,96 @@ export const PatientManager = (props: any) => {
...
+          }
         >
           <div className="px-4 md:w-1/2">
             <div className="md:flex justify-between w-full">
               <div className="text-xl font-normal capitalize">
                 {patient.name} - {patient.age}
-                {
-                  patient.action && patient.action != 10 && <span className="font-semibold ml-2">- {TELEMEDICINE_ACTIONS.find(i => i.id === patient.action)?.desc}</span>
-                }
+                {patient.action && patient.action != 10 && (
```
Comment (riginoommen) [inline src/Components/Patient/ManagePatients.tsx]:
```suggestion
                {patient.action && patient.action !== 10 && (
```

File: src/Components/Patient/PatientFilterV2.tsx [hunk 16]
```diff
@@ -1,7 +1,7 @@
 import React, { useEffect, useState } from "react";
 import { FacilitySelect } from "../Common/FacilitySelect";
 import { SelectField, DateInputField, TextInputField } from "../Common/HelperInputFields";
```
Comment (riginoommen) [inline src/Components/Patient/PatientFilterV2.tsx]:
```suggestion
import React, { useEffect, useState } from 'react;
import { FacilitySelect } from '../Common/FacilitySelect';
import { SelectField, DateInputField, TextInputField } from '../Common/HelperInputFields';
```

File: src/Components/Patient/PatientFilterV2.tsx [hunk 17]
```diff
@@ -1,7 +1,7 @@
...
-import { PATIENT_FILTER_ORDER, GENDER_TYPES, DISEASE_STATUS, PATIENT_FILTER_CATEGORY, PATIENT_FILTER_ADMITTED_TO } from "../../Common/constants";
+import { PATIENT_FILTER_ORDER, GENDER_TYPES, DISEASE_STATUS, PATIENT_FILTER_CATEGORY, PATIENT_FILTER_ADMITTED_TO, PATIENT_FILTER_DECLARED_STATUS } from "../../Common/constants";
```
Comment (riginoommen) [inline src/Components/Patient/PatientFilterV2.tsx:4]:
```suggestion
import { PATIENT_FILTER_ORDER, GENDER_TYPES, DISEASE_STATUS, PATIENT_FILTER_CATEGORY, PATIENT_FILTER_ADMITTED_TO, PATIENT_FILTER_DECLARED_STATUS } from '../../Common/constants';
```

File: src/Components/Patient/PatientFilterV2.tsx [hunk 18]
```diff
@@ -1,7 +1,7 @@
...
 import moment from "moment";
 import { getFacility } from '../../Redux/actions';
 import { useDispatch } from 'react-redux';
```
Comment (riginoommen) [inline src/Components/Patient/PatientFilterV2.tsx]:
```suggestion
import moment from 'moment';
import { getFacility } from '../../Redux/actions';
import { useDispatch } from 'react-redux';
```

File: src/Components/Patient/PatientFilterV2.tsx [hunk 19]
```diff
@@ -30,6 +30,9 @@ export default function PatientFilterV2(props: any) {
     disease_status: filter.disease_status || null,
     age_min: filter.age_min || null,
     age_max: filter.age_max || null,
+    is_declared_positive: filter.is_declared_positive || null,
+    date_declared_positive_before: filter.date_declared_positive_before || null,
+   date_declared_positive_after: filter.date_declared_positive_after || null,
```
Comment (riginoommen) [inline src/Components/Patient/PatientFilterV2.tsx:59]:
```suggestion
    date_declared_positive_after: filter.date_declared_positive_after || null,
```

File: src/Components/Patient/ManagePatients.tsx [hunk 20]
```diff
@@ -1,20 +1,20 @@
-import loadable from '@loadable/component';
-import Box from '@material-ui/core/Box';
+import loadable from "@loadable/component";
+import Box from "@material-ui/core/Box";
 import Button from "@material-ui/core/Button";
 import Grid from "@material-ui/core/Grid";
```
Comment (riginoommen) [inline src/Components/Patient/ManagePatients.tsx]:
```suggestion
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
```

**Review by riginoommen — CHANGES_REQUESTED** [review]

**Comment by smartclash** [comment]
Hey @7tena, which issue are you trying to fix in this PR?

File: debug.log [hunk 21]
```diff
@@ -0,0 +1 @@
+[1122/181517.624:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
```
Comment (smartclash) [inline debug.log:1]: This file should not exist.

Comment (smartclash) [inline debug.log:1]: This file should not exist :)

**Review by smartclash — CHANGES_REQUESTED** [review]

File: debug.log (hunk 21 above)
Comment (7tena) [reply debug.log:1]: The issue got fixed by someone else. Don't mind the PR.

**Comment by smartclash** [comment]
Thank you for making a PR @7tena, do join our slack channel so the response from other peer developers will helpful. http://slack.coronasafe.in/ 

Closing this since the issue has been fixed.

**Pull request closed without merging**